go/sqltypes/proto3_test.go

```diff
@@ -35,6 +35,13 @@ func TestResult(t *testing.T) {
 			NULL,
 			NULL,
 		}},
+		Extras: &querypb.ResultExtras{
```

Comment:
It actually doesn't match ExecuteOptions because they affect field info, and may affect other parts of the result in the future.
The reason why I question Extras is because the name sounds like a catch-all, and doesn't provide a clear categorization. One may argue that Fields is also an extra. In my eyes, everything other than Rows (and field types) are extra.
Other questions:
* Will it improve storage/network efficiency?
* Will it improve readability: do we need to pass Extras around separately as a parameter?